ReactiveCocoaTests/Swift/LifetimeSpec.swift

```diff
@@ -42,27 +42,25 @@ final class LifetimeSpec: QuickSpec {
 		}
 
 		describe("NSObject lifetime") {
-			it("ends when the object is deallocated") {
+			it("should complete its signal even if it is being retained") {
 				let object = MutableReference(TestObject())
+				let lifetime = object.value!.lifetime
 
-				var events: [String] = []
-
-				object.value!.lifetime.ended.observe { event in
-					switch event {
-					case .next:
-						events.append("next")
-					case .completed:
-						events.append("completed")
-					case .failed:
-						events.append("failed")
-					case .interrupted:
-						events.append("interrupted")
-					}
-				}
+				var isCompleted = false
+				lifetime.ended.observeCompleted { isCompleted = true }
 
 				object.value = nil
```

Comment:
I think it mostly makes it a bit clearer that you're testing that `object.value = nil` is expected to trigger `isCompleted = true`. But I can envision a bug where the completed block could complete immediately.